homeassistant/components/ssdp/__init__.py

```diff
@@ -85,6 +88,65 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+@dataclass
+class _HaServiceDescription:
+    """Keys added by HA."""
+
+    x_homeassistant_matching_domains: set[str]
+
+
+@dataclass
+class _SsdpServiceDescription:
+    """SSDP info with optional keys."""
+
+    ssdp_usn: str
+    ssdp_st: str
+    ssdp_location: str | None = None
+    ssdp_nt: str | None = None
+    ssdp_udn: str | None = None
+    ssdp_ext: str | None = None
+    ssdp_server: str | None = None
+
+
+@dataclass
+class _UpnpServiceDescription:
+    """UPnP info."""
+
+    upnp: dict[str, Any]
```

Comment:
A default should not be necessary, but it might make it easier when we create the test instances (some components don't use it).
```suggestion
    upnp: dict[str, Any] = field(default_factory=dict)
```

Reply:
This one we probably need to think about in tests so we should probably make it required

Reply:
OK, I'll leave in.